### ISSUE TITLE
Reloading namespace with REPL request invalid URLs for goog library

### ISSUE THREAD

**Issue by hura** [bhauman/lein-figwheel#116]
I have many of these errors in my console when I `cider-eval-buffer` a file that has `cljs-time` as a `(:require `:

```
GET http://localhost:3449/js/out/goog/date/DateLike.js?zx=1yhr1il6vrg8
goog.net.jsloader.load **jsloader.js:193**
(anonymous function) **file_reloading.cljs:87**
cljs.core.MultiFn.call.G__11384__3 **core.cljs:9270**
cljs.core.MultiFn.call.G__11384 **core.cljs:9379**
figwheel.client.file_reloading.reload_file_STAR_.figwheel$client$file_reloading$reload_file_STAR___2 **file_reloading.cljs:93**
figwheel.client.file_reloading.reload_file_STAR_.figwheel$client$file_reloading$reload_file_STAR_ **file_reloading.cljs:93**
figwheel.client.file_reloading.reload_file_STAR_.figwheel$client$file_reloading$reload_file_STAR___1 **file_reloading.cljs:94**
figwheel.client.file_reloading.reload_file_STAR_.figwheel$client$file_reloading$reload_file_STAR_ **file_reloading.cljs:93**
figwheel.client.file_reloading.patch_goog_base.goog.require **file_reloading.cljs:59**
(anonymous function) **VM5065:1**
(anonymous function) **repl.cljs:28**
(anonymous function) **repl.cljs:37**
cljs.core.MultiFn.call.G__11384__2 **core.cljs:9264**
cljs.core.MultiFn.call.G__11384 **core.cljs:9250**
(anonymous function) **repl.cljs:70**
goog.events.EventTarget.fireListeners **eventtarget.js:285**
goog.events.EventTarget.dispatchEventInternal_ **eventtarget.js:382**
goog.events.EventTarget.dispatchEvent **eventtarget.js:197**
goog.net.WebSocket.onMessage_ **websocket.js:411**
```

All errors:
```
GET http://localhost:3449/js/out/goog/date/DateLike.js?zx=1yhr1il6vrg8
GET http://localhost:3449/js/out/goog/date.js?zx=jc6ks0r7ar48 
GET http://localhost:3449/js/out/goog/date/UtcDateTime.js?zx=pmjsy17vu35z 
GET http://localhost:3449/js/out/goog/i18n/TimeZone.js?zx=j8ne4y5fvsdx 
GET http://localhost:3449/js/out/goog/string/format.js?zx=lfvymek80kxu 
core.cljs:222Uncaught TypeError: Cannot read property 'Date' of undefined
GET http://localhost:3449/js/out/goog/date/DateTime.js?zx=e19jynq8he1h 
GET http://localhost:3449/js/out/goog/date/Date.js?zx=cvayxthu4376 
```

That's a repl started with `lein repl`. Actually, I'm not sure why figwheel interferes with a `lein repl` started nREPL and only cider middleware.  The figwheel reloading works fine, just not when I evaluate the entire (emacs) buffer through the REPL.

I'm sorry if this is a silly question. I'm new(ish) to the clojurescript world.

**Comment by bhauman** [comment]
There is no where near enough context here for me to know what you are referring to.  You are running a clojureScript repl on top of the lein repl correct?  What kind of repl? is it figwheel's repl? is it weasel? is it the cljs Browser repl?

**Comment by hura** [comment]
I'm running the reagen template:
https://github.com/reagent-project/reagent-template
which is basically the same as a chestnut template.

My project.clj: https://gist.github.com/hura/c08ae46f91d65f287fd8

I start my repl with `lein repl` and then start my browser REPL with:
```clj
(defn browser-repl []
  (piggieback/cljs-repl
   :repl-env
   (weasel/repl-env
    :ip "localhost"
    :port 9001)))
```

I connect my Emacs with cider-connect to that REPL and type `(browser-repl)`. The browser connects through the fighweel server that's started with a `lein figwheel`.

Btw, all the "GET" errors because the case of the requested is wrong:

http://localhost:3449/js/out/goog/date/DateLike.js (404)
http://localhost:3449/js/out/goog/date/datelike.js (exists)

Thank you for your help.

**Comment by abhishiv** [comment]
I'm running into this issue after upgrading to 0.2.5. Some context in case it helps.

My project.clj looks like this
```
  :cljsbuild {
    :builds {
      :gdesigner {
        :source-paths ["src/cljs/satya" "src/cljs/gratico/common" "src/cljs/gratico/designer"]
        :compiler {
          :output-to     "resources/public/js/gdesigner.js"
          :output-dir    "resources/public/js/gdesigner"
          :source-map    "resources/public/js/gdesigner.js.map"
          :optimizations :none
          :externs ["externs/noJSX.js"]
          :pretty-print  true}}}}

```

So the goog.addDependecy in `resources/public/js/gdesigner.js`  call looks like this `goog.addDependency("../gratico/designer/orkester/maya.js", ['gratico.designer.orkester.maya'], ['cljs.core', 'cljsjs.react', 'cljsjs.react_router']);` resulting into script tag like this `<script type="text/javascript" src="http://localhost:10555/js/gdesigner/goog/../gratico/designer/orkester/maya.js"></script>`

Now when I change this file, there's a 404 to this url http://localhost:3449/gratico/designer/orkester/maya.js?zx=1zr1jgdps42m 

So for some reason, it's not appending `js/gdesigner/goog/..`

**Comment by abhishiv** [comment]
Oh, lein cljsbuild clean fixed it. Should have done it after the upgrade. Thanks!

**Comment by hura** [comment]
I have some additional info (now that I got it to work):

I didn't have my namespace that the error was from (klang.core) required in my main app so it was never referenced.
Once I added a simple `(require ..` to my main application file the error was also triggered in a figwheel reload. I then restarted figwheel and everything (including the repl) works now.

I guess this could be considered a user error. However, could also be room for improvement to figwheel. Feel free to close this.
